### PR TITLE
New cli version adaptation CERT-1988

### PR DESCRIPTION
**Pull request by NoaBablikiCertora**
* adapt the extension for the new cli version
* added error handling for when a job cannot run
* get the rule report link from .certora_internal - IN PROGRESS

### CODE REVIEW

**Comment by NoaBablikiCertora** [comment]
get the rule report link from .certora_internal - IN PROGRESS - in XXXX ticket (in the future)
INSTEAD: get the new link from the logs

**Review by Biced — APPROVED** [review]
Corgi keshet!
Happy #100!!!!!!!!

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 All notable changes to this project will be documented in this file.
 
+### 0.1.8 ()
```
Comment (sarittenen) [inline CHANGELOG.md:5]: add date

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,13 @@
...
+
+## Features
+
+* Compatibility with the newest certora-cli version ([#100](https://github.com/Certora/vscode-certora-prover/pull/98))
```
Comment (sarittenen) [inline CHANGELOG.md:9]: please replace "the newest" with "latest"

File: src/extension/ScriptRunner.ts [hunk 3]
```diff
@@ -128,25 +130,48 @@ export class ScriptRunner {
...
-  }
-
   /**
    * get the job id from the link to the verification report
    * @param link link to the verification report (string)
    * @returns job id (string)
    */
   private getJobId(link: string): string {
-    const pattern = 'https://(prover|vaas-stg).certora.com/output/'
+    const pattern = 'https://(prover|vaas-stg).certora.com/(output|jobStatus)/'
```
Comment (sarittenen) [inline src/extension/ScriptRunner.ts:139]: is this still necessary? (after url from json fix)

File: src/extension/ScriptRunner.ts [hunk 4]
```diff
@@ -128,25 +130,48 @@ export class ScriptRunner {
...
     const regExp = new RegExp(pattern)
     return link.split('?anonymousKey')[0].replace(regExp, '').split('/')[1]
   }
 
+  /**
+   * check the cli-version
+   * @param confFile path to conf file (string)
+   * @returns true if cli version is found, false otherwise
+   */
+  private checkVersion(confFile: string) {
```
Comment (sarittenen) [inline src/extension/ScriptRunner.ts:149]: -> checkCliVersion

File: src/extension/ScriptRunner.ts [hunk 5]
```diff
@@ -315,7 +409,7 @@ export class ScriptRunner {
       }
 
       fetch(
-        scriptToStop.vrLink?.split('/output/')[0] +
+        scriptToStop.vrLink?.split('/(output|jobStatus)/')[0] +
```
Comment (sarittenen) [inline src/extension/ScriptRunner.ts:412]: same question

File: src/extension/utils/getProgressUrl.ts [hunk 6]
```diff
@@ -10,7 +10,11 @@ export function getProgressUrl(text: string): string | null {
     const url = urlMatches.find(url => {
       return url.includes('anonymousKey')
     })
-    return url?.includes('output') ? url.replace('output', 'progress') : null
+    if (url?.includes('output')) {
```
Comment (sarittenen) [inline src/extension/utils/getProgressUrl.ts:13]: same question 3 :)

File: src/settings/utils/confFileToFormData.ts [hunk 7]
```diff
@@ -263,30 +268,33 @@ export function confFileToFormData(confFile: ConfFile): NewForm {
       }
     }
   }
-
+  // confFile.verify can either be an array or a string
```
Comment (sarittenen) [inline src/settings/utils/confFileToFormData.ts:271]: are you checking old cli ver - array, new cli - string?

**Pull request merged**